src/node_options-inl.h

```diff
@@ -0,0 +1,421 @@
+#ifndef SRC_NODE_OPTIONS_INL_H_
+#define SRC_NODE_OPTIONS_INL_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include "node_options.h"
+#include "util.h"
+
+namespace node {
+
+PerIsolateOptions* PerProcessOptions::get_per_isolate_options() {
+  return per_isolate.get();
+}
+
+DebugOptions* EnvironmentOptions::get_debug_options() {
+  return debug_options.get();
+}
+
+EnvironmentOptions* PerIsolateOptions::get_per_env_options() {
+  return per_env.get();
+}
+
+template <typename Options>
+void OptionsParser<Options>::AddOption(const std::string& name,
+                                       bool Options::* field,
+                                       OptionEnvvarSettings env_setting) {
+  options_.emplace(name, OptionInfo {
+    kBoolean,
```

Comment:
Just thinking out loud... (It is an interesting problem after all).
How about instead on a `const` you put a pointer to the conversion function. That way you make these Templates specializations on the `typeof(field)`, and eliminate the switch in L385.

Reply:
Had a few more "mind itches". I'll try to write up a draft.